### ISSUE TITLE
Can't get lookahead to work

### ISSUE THREAD

**Issue by mateomurphy** [rmosolgo/graphql-ruby#2569]
I'm having trouble getting lookahead working, and I'm not sure if I'm doing something wrong or if there's a bug. I've set up a test project using version 1.9.14, with a post type as described at https://graphql-ruby.org/getting_started but changed the QueryType as follows:

```
module Types
  class QueryType < Types::BaseObject
    description "The query root of this schema"

    # First describe the field signature:
    field :post, PostType, null: true, extras: [:lookahead] do
      description "Find a post by ID"
      argument :id, ID, required: true
    end

    # Then provide an implementation:
    def post(lookahead:, id:)
      Post.find(id)
    end
  end
end
```

However when I execute a query I get a `ArgumentError: missing keyword: lookahead`, as `lookahead` is always `nil`. Am I doing something wrong?

**Comment by rmosolgo** [comment]
Hi, sorry about the hangup! I'm not sure what's wrong. It would help if you shared the full stack trace, because that would show me what graphql-ruby was doing when it encountered an error. 

I tried to replicate the issue with this script, but it ran properly for me: 

```ruby 
require "bundler/inline"

gemfile do
  source 'https://rubygems.org'
  gem "graphql", "1.9.14"
end

puts GraphQL::VERSION

class Query < GraphQL::Schema::Object
  field :str, String, null: false, extras: [:lookahead] do
    argument :input, String, required: true
  end

  def str(lookahead:, input:)
    input + " + " + lookahead.class.name
  end
end

class Schema < GraphQL::Schema
  query(Query)
end

p Schema.execute('{ str(input: "abc") }')
# => {"data"=>{"str"=>"abc + GraphQL::Execution::Lookahead"}}
```

I'm happy to debug this further, but I don't have anything to go on. If you'd like to keep looking at it please do one or both of: 

- Add the stack trace to this bug report; AND/OR 
- Update the script above to demonstrate the bug. 

Then reopen this issue, and I'll take another look!

**Comment by mateomurphy** [comment]
Hi, I've recreated the issue with the following code

```rb
require "bundler/inline"

gemfile do
  source 'https://rubygems.org'
  gem "graphql", "1.9.14"
end

puts GraphQL::VERSION

class BaseField < GraphQL::Schema::Field
  def resolve_field(obj, args, ctx)
    resolve(obj, args, ctx)
  end
end

class BaseObject < GraphQL::Schema::Object
  field_class BaseField
end

class Query < BaseObject
  field :str, String, null: false, extras: [:lookahead] do
    argument :input, String, required: true
  end

  def str(lookahead:, input:)
    input + " + " + lookahead.class.name
  end
end

class Schema < GraphQL::Schema
  query(Query)
end

p Schema.execute('{ str(input: "abc") }')
# => {"data"=>{"str"=>"abc + GraphQL::Execution::Lookahead"}}
```

The issue is with the `BaseField` class, which is generated by `rails g graphql:install`: it works properly when I remove the `resolve_field` method

**Comment by rmosolgo** [comment]
> reopen issues close by collaborators, afaik

😖 I'm sorry, I didn't realize that. I'll have to be more careful with that. 

Thanks for sharing a reproduction. That `def resolve_field` snuck in in #2216 and I have no idea why. I think it's basically bad code. 

I'll try to find the problem here and release a fix ... _and_ fix that generator!

**Comment by mateomurphy** [comment]
No worries. I have a work around for now. Thanks for the help!

**Comment by rmosolgo** [comment]
I would _expect_ everything to work fine if you remove the `def resolve_field` method from the generated base field class. Does that work ok for you? I have no idea why that's there 😖

**Comment by mateomurphy** [comment]
Yes, it works fine when that method is removed

**Comment by rmosolgo** [comment]
Glad to hear it. I don't have plans to support that wacky code in the generated base field, but I'll fix the generator in 1.9.15. Thanks for your help debugging!
